scripts/temporary_test_read_records.py

```diff
@@ -0,0 +1,57 @@
+from jiant.utils import serialize
```

Comment:
Can you please make this into a unit test?

Reply:
I think this is more like a regression test. I don't see an easy way to convert it into a unit test.

Reply:
Hey @HaokunLiu — I tried to write something similar to the tests you wrote, but tried to make them more like unit tests. These tests end up involving `write_records`, but they'll add some protection from changes going forward: 

```
class TestReadRecords(unittest.TestCase):

    def test_read_records_without_data_fraction(self):
        """write records then read records (with no data fraction arg), check that records match"""
        data_file_name = "data.pb64"
        fake_example_count = 100
        fake_examples = [i for i in range(fake_example_count)]
        with tempfile.TemporaryDirectory() as tmp_dir_path:
            fake_filepath = os.path.join(tmp_dir_path, data_file_name)
            serialize.write_records(fake_examples, fake_filepath)
            fake_examples_read = serialize.read_records(fake_filepath)
            self.assertCountEqual(fake_examples, fake_examples_read)

    def test_read_records_with_data_fraction(self):
        """write examples, read without fraction, read with fraction, check expected files exist"""
        filename = "data.pb64"
        frac = 0.1
        fake_example_count = 100
        fake_examples = [i for i in range(fake_example_count)]
        with tempfile.TemporaryDirectory() as tmp_dir_path:
            fake_filepath = os.path.join(tmp_dir_path, filename)
            serialize.write_records(fake_examples, fake_filepath)
            fake_examples_read = serialize.read_records(fake_filepath)
            fake_examples_read_frac = serialize.read_records(fake_filepath, fraction=frac)
            file_list = os.listdir(tmp_dir_path)
            self.assertLess(len(list(fake_examples_read_frac)), len(list(fake_examples_read)))
            self.assertTrue(set(fake_examples_read_frac).issubset(set(fake_examples_read)))
            self.assertCountEqual(file_list, [filename, filename + "__fraction_" + str(frac)])

    def test_read_records_with_data_fraction_without_prior_full_data_read(self):
        """write examples, then read with fraction (no full read before fractional read)"""
        filename = "data.pb64"
        frac = 0.1
        fake_example_count = 100
        fake_examples = [i for i in range(fake_example_count)]
        with tempfile.TemporaryDirectory() as tmp_dir_path:
            fake_filepath = os.path.join(tmp_dir_path, filename)
            serialize.write_records(fake_examples, fake_filepath)
            fake_examples_frac = serialize.read_records(fake_filepath, fraction=frac)
            self.assertLess(len(list(fake_examples_frac)), len(list(fake_examples)))
            self.assertTrue(set(fake_examples_frac).issubset(set(fake_examples)))

    def test_read_records_with_data_fraction_from_cache(self):
        """write examples, read with fraction, then read with fraction again (from cache)"""
        filename = "data.pb64"
        frac = 0.1
        fake_example_count = 100
        fake_examples = [i for i in range(fake_example_count)]
        with tempfile.TemporaryDirectory() as tmp_dir_path:
            fake_filepath = os.path.join(tmp_dir_path, filename)
            serialize.write_records(fake_examples, fake_filepath)
            fake_examples_frac = serialize.read_records(fake_filepath, fraction=frac)
            os.remove(fake_filepath)  # remove full file make sure we get frac examples from cache
            fake_examples_frac_cache = serialize.read_records(fake_filepath, fraction=frac)
            self.assertCountEqual(list(fake_examples_frac), list(fake_examples_frac_cache))
```